### ISSUE TITLE
How to hardcode only 1 containter!?

### ISSUE THREAD

**Issue by FractalMind** [evertramos/docker-compose-letsencrypt-nginx-proxy-companion#28]
Excellent job guys! I'm very impressed with your work!

**Where in the code I could modify the internal router that route the SNI _website1.com TO container:website1 =>(VIRTUAL_HOST=website1.com)_ so I can hardcode all traffic to the same container:localhost???**

I'm developing an app like "wix.com, squarespace.com" to generate website on the fly with HTTPS support. This project is the closest I've found to meet our criteria to solve our TLS termination problems.

The only dilemma is this project was designed for websites with different code
EX: 
website1.com => custom code for website1
website2.com => custom code for website2
Each websites are separated with their own code; which is logical. 

The problem is our app is already polymorphic(will pull the custom data according to the hostname)
EX:
website1.com => uses the SAME code to build website1 with DIFFERENT data according to the hostname.
website2.com => uses the SAME code to build website2 with DIFFERENT data according to the hostname.

**Long story short:** We cannot have 1,000,000 containers: website1,website2,...website100001; it makes no sense. It's all the same code! We just need "on the fly TLS termination". I hope to decrypt/generate all SSL on the fly and bump all the traffic to our code to do the rest of the magic.

**Where in the code I could modify the internal router that route the SNI _website1.com TO container:website1 =>(VIRTUAL_HOST=website1.com)_ so I can hardcode all traffic to the same container:localhost???**

Thank you for your time :)

**Comment by evertramos** [comment]
Hello @FractalMind!

As far as I understood you want to add multiples domain into one container using the same code, is that it?

I would suggest two things:

1. You can add _n_ host into `VIRTUAL_HOST=website1.com,website2.com,...website10000.com`
 
Might suit your needs.

But if you need a little more management on data if there is any different data for any client, you would sepparate a few things, using volumes, which would be my second option:

2. After get webproxy running, you create this folder structure:

```
- sites /
  | - data / {which would be the basic data for all websites
  |  - website1.com / 
     | - data / {specific data if there is any}
  |  - website2.com / 
     | - data / {specific data if there is any}
  |  - website1000.com / 
     | - data / {specific data if there is any}
```

Make all of your containers read the ./site/data folder as your base data, if possible using read only (:ro)

In that case, yes you would have one container for each website, but that´s the main idea of containers, isn´t it? And if one of your clients/user make some very big thing it would probably drop down the container... if you have only one container, might get all clients off line at same time.

Not sure if my answer would help you out for sure... but just give a try!

Happy New Year!!

**Comment by FractalMind** [comment]
Thank you for your quick answer @evertramos !

Your solution is very interesting but there are multiple problems with this approach:
1- Not dynamic: Every time I have to create a new site, I doubt I can just update the container. I'll have to delete the running container, append the new domain to VIRTUAL_HOST=[...], website232434334.com, rerun the new container without downtime. I might also run into parameters limitations. It's too much gymnastics and not an elegant solution.
2- Scale: I think it's important to keep in mind that we might have a million websites running not 15.

TSL Termination. That's basically all we need. As soon as the request is decrypted we are good.

Code: Our code is polymorphic. We don't need a (sites) folder because we pull everything from the DB and stitch together the site on the fly. Meaning. The code is universal and identical no matter the domain.

The "perfect" solution would be: Create a container with VIRTUAL_HOST=localhost, modify the proxy or the "let's encrypt" module to redirect all decrypted traffic to the one (localhost: container). As long as the HTTP_HOST: EX: website1212.com is accessible in the container we are good :)

Could you point me where I could make these modifications?

Thank you for your time :)

**Comment by evertramos** [comment]
@FractalMind! Happy New Year!

Well I am not that wise... the "brains" that create it was @jwilder and @JrCs, they probably will be able to help you out.

But I think, in the number 1 suggestions, you could use and `.env` file updated dinamically by your app, which probably it´s easy to set up, and just update the running compose file with a simple `docker-compose up -d` command, and it will automatically update the dns and create ssl certificates etc.

Hope it will help. I will close this as an issue but we can continue to discuss and I really would like to know what will be your final environment.

Cheers!
